tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/PublicApiInstanceMetadataTests.cs

```diff
@@ -0,0 +1,146 @@
+                    .Create(TestConstants.ClientId)
+                    .WithAuthority(new Uri(@"https://sts.windows.net/common/"), false)
+                    .WithInstanceDicoveryMetadata(instanceMetadataJson)
+                    .WithHttpManager(harness.HttpManager)
+                    .WithTelemetry(new TraceTelemetryConfig())
+                    .BuildConcrete();
+
+                var ex = await Assert.ThrowsExceptionAsync<MsalClientException>(() => app
+                    .AcquireTokenInteractive(TestConstants.s_scope)
+                    .ExecuteAsync(CancellationToken.None))
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual(MsalError.InvalidUserInstanceMetadata, ex.ErrorCode);
+            }
+        }
+
+        [TestMethod]
+        [DeploymentItem(@"Resources\CustomInstanceMetadata.json")]
+        public async Task CustomInstanceDiscoveryMetadataUri_Async()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                var customMetadataUri = new Uri("https://custom.instance.discovery.uri");
+                var customAuthrority = "https://my.custom.authority/common/";
+
+                harness.HttpManager.AddInstanceDiscoveryMockHandler(
+                    customAuthrority, 
+                    customMetadataUri, 
+                    TestConstants.DiscoveryJsonResponse.Replace("login.microsoftonline.com", "my.custom.authority"));
+
+                PublicClientApplication app = PublicClientApplicationBuilder
+                    .Create(TestConstants.ClientId)
+                    .WithAuthority(customAuthrority, false)
+                    .WithInstanceDicoveryMetadata(customMetadataUri)
+                    .WithHttpManager(harness.HttpManager)
+                    .BuildConcrete();
+
+                MsalMockHelpers.ConfigureMockWebUI(
+                    app.ServiceBundle.PlatformProxy,
+                    AuthorizationResult.FromUri(app.AppConfig.RedirectUri + "?code=some-code"));
+
+                // the rest of the communication with AAD happens on the preferred_network alias, not on login.windows.net
+                harness.HttpManager.AddMockHandlerForTenantEndpointDiscovery(customAuthrority);
+                harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(customAuthrority);
+
+                AuthenticationResult result = await app
+                    .AcquireTokenInteractive(TestConstants.s_scope)
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(result);
+                Assert.IsNotNull(result.Account);
+                Assert.AreEqual(TestConstants.UniqueId, result.UniqueId);
+                Assert.AreEqual(TestConstants.CreateUserIdentifier(), result.Account.HomeAccountId.Identifier);
+                Assert.AreEqual(TestConstants.DisplayableId, result.Account.Username);
```

Comment:
do you need to assert the endpoint values?

Reply:
I think the HttpManager ensures that the `customMetadataUri ` is hit.